x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessHeadersForCcsRestIT.java

```diff
@@ -0,0 +1,506 @@
+                            new RoleDescriptor(
+                                RBACEngine.REMOTE_USER_ROLE_NAME,
+                                null,
+                                new RoleDescriptor.IndicesPrivileges[] {
+                                    RoleDescriptor.IndicesPrivileges.builder()
+                                        .indices("index-a")
+                                        .privileges("read", "read_cross_cluster")
+                                        .build(),
+                                    RoleDescriptor.IndicesPrivileges.builder()
+                                        .indices("index-b")
+                                        .privileges("read", "read_cross_cluster")
+                                        .build() },
+                                null,
+                                null,
+                                null,
+                                null,
+                                null,
+                                null
+                            )
+                        )
+                    )
+                )
+            );
+        }
+    }
+
+    private void setupClusterSettings(
+        final String clusterAlias,
+        final String clusterCredential,
+        final DiscoveryNode remoteNode,
+        boolean useProxyMode
+    ) throws IOException {
+        if (useProxyMode) {
+            updateRemoteClusterSettings(
+                clusterAlias,
+                Map.of("mode", "proxy", "proxy_address", remoteNode.getAddress().toString(), "authorization", clusterCredential)
+            );
+        } else {
+            updateRemoteClusterSettings(
+                clusterAlias,
+                Map.of("seeds", remoteNode.getAddress().toString(), "authorization", clusterCredential)
+            );
+        }
+    }
+
+    private void expectActionsAndHeadersForCluster(
+        final List<CapturedActionWithHeaders> actualActionsWithHeaders,
+        boolean useProxyMode,
+        boolean minimizeRoundtrips,
+        final String clusterCredential,
+        final RoleDescriptorsIntersection expectedRoleDescriptorsIntersection
+    ) throws IOException {
+        final Set<String> expectedActions = new HashSet<>();
+        if (minimizeRoundtrips) {
+            expectedActions.add(SearchAction.NAME);
+        } else {
+            expectedActions.add(ClusterSearchShardsAction.NAME);
+        }
+        if (false == useProxyMode) {
+            expectedActions.add(ClusterStateAction.NAME);
+        }
+        assertThat(
+            actualActionsWithHeaders.stream().map(CapturedActionWithHeaders::action).collect(Collectors.toUnmodifiableSet()),
+            equalTo(expectedActions)
+        );
+        for (CapturedActionWithHeaders actual : actualActionsWithHeaders) {
+            switch (actual.action) {
+                // the cluster state action is run by the system user, so we expect an authentication header, instead of remote access
```

Comment:
We will need figure out how to support this, but again due to scope, this will be a separate PR -- one where we tackle internal users.